lib/bank_accounts/account.rb

```diff
@@ -0,0 +1,51 @@
+module BankAccount
+
+  class Account
+    attr_accessor :id, :balance
+#_________________________________________________________________________________________________
+# initialization
+
+    def initialize(id, initial_balance)
+      # # Checking if id is a number
+      # # if it's not then it raises ArgumentError
+      if id.class != Fixnum
+        raise ArgumentError
+      end
+
+      @id = id
+
+      # Checking if initial_balance is a positive number
+      # if it's not then it raises ArgumentError
+      if (initial_balance.class != Fixnum && initial_balance.class != Float) || initial_balance < 0
```

Comment:
Most of the time, you should do all of your error checking at the same time; verify before you do any other work. I like that you are verifying the class of your input and raising an error, but don't do any work (assigning `@id` is work) until you have validated all your input.